### ISSUE TITLE
Recent alsa-ucm-conf update requires ucm syntax 7

### ISSUE THREAD

**Issue by tails1154** [WeirdTreeThing/chromebook-linux-audio#185]
**Describe the bug**
I literately just did `sudo pacman -Syu` and rebooted and the sound is broken now

**Boardname**
CROTA

**Logs**
[debug-logs-Crota-2024-11-25_18h19m.tar.gz](https://github.com/user-attachments/files/17910409/debug-logs-Crota-2024-11-25_18h19m.tar.gz)

**Comment by mikezila** [comment]
I'm having the same issue on LEONA. My issue started after a kernel update on Arch. Alsa and friends were updated the next day and it did not fix it. I've re-run the script and it is still not functional.

I'm not sure what type of log would be most helpful, but here's dmesg:

[log.txt](https://github.com/user-attachments/files/17910782/log.txt)

**Comment by tails1154** [comment]
@mikezila 

run `wget https://raw.githubusercontent.com/chrultrabook/linux-tools/main/debugging.sh; bash debugging.sh`  and upload the file

**Comment by Integral-Tech** [comment]
Same problem for me.

**Comment by slimx** [comment]
[debug-logs-Kled-2024-11-26_17h51m.tar.gz](https://github.com/user-attachments/files/17917440/debug-logs-Kled-2024-11-26_17h51m.tar.gz)

**Comment by 2DSBrewer** [comment]
Same here. Updated Arch today and broken audio.
[debug-logs-Dragonair-2024-11-26_09h25m.tar.gz](https://github.com/user-attachments/files/17920978/debug-logs-Dragonair-2024-11-26_09h25m.tar.gz)

**Comment by ninelore** [comment]
Y'all tried re-running the script?

**Comment by 2DSBrewer** [comment]
Yes, or at least I did.

**Comment by ninelore** [comment]
Just making sure nothing silly on arch side did touch ucm.

Smells like an upstream alsa issue to me

**Comment by tails1154** [comment]
yea I have no idea what yall are talking about but
1. I did rerun the script, did nothing
2. when I went to do `pacman -Syu` it complained about some conflicting file, I made a copy of it and deleted it, rebooted. No sound. I put my backup file back there and rebooted. Nothing.
## I hope this gets fixed

**Comment by tails1154** [comment]
(i think) alsa just got a update. I'm updating now

**Comment by tails1154** [comment]
nope did not fix anything.

**Comment by slimx** [comment]
`downgrade alsa-ucm-profile` to version 2.12.

**Comment by mikaelvz** [comment]
I'm on Fedora 41 :
- with latest commit, no sound, there is only the dummy device
- with previous commit (2002c72e108c99b822d94e1b383f68a00d7cca9d), I have sound

**Comment by WeirdTreeThing** [comment]
The issue most of you seem to have is related to the dmic split. I was just able to reproduce this issue myself and found that removing the dmic split fixes it. I'll look into how to properly fix this issue.

@mikaelvz What board name? Must be one where the dmic split isn't needed.

**Comment by mikaelvz** [comment]
> @mikaelvz What board name? Must be one where the dmic split isn't needed.
@WeirdTreeThing sorry, forget to mention it, I'm on KLED board

**Comment by CoolPenguin27** [comment]
I can confirm that using the `syntax-7` branch fixed the issue for me.

**Comment by tails1154** [comment]
let me try that

**Comment by Castaevsky** [comment]
"./setup-audio --branch syntax-7" doesn't work on chromebook hp g7 11 ee,with archlinux installed within

**Comment by mikaelvz** [comment]
Using syntax-7 branch also work for me with on KLED, thanks.

**Comment by mikezila** [comment]
The "branch-7" fix mentioned above did not work for LEONA.

**Comment by WeirdTreeThing** [comment]
leona doesnt use the dmic split so it would have a separate issue

**Comment by tails1154** [comment]
ok I will reopen it I guess

**Comment by 2DSBrewer** [comment]
> Looks like the fix is to increase the syntax version. Try `./setup-audio --branch syntax-7` and see if that works.

Yup. On dragonair it works now.

**Comment by dupsneerg** [comment]
No joy with Arch on SNAPPY.
`sudo pacman -Syu` upgraded the kernel to 6.12 after I removed the conflicting file similar to @tails1154 mentioned.
I downgraded alsa-ucm-conf to version 2.12.
I reran the script, adding the flag mentioned above: `./setup-audio --branch syntax-7` using the sof option.
I only have dummy output when checking `pavucontrol`.
I ran the diagnostic script mentioned above: `wget https://raw.githubusercontent.com/chrultrabook/linux-tools/main/debugging.sh; bash debugging.sh` and added its output.  Hoping that the extra info helps troubleshoot.  I appreciate the work--it's above my paygrade!
[debug-logs-Snappy-2024-12-05_12h59m.tar.gz](https://github.com/user-attachments/files/18027675/debug-logs-Snappy-2024-12-05_12h59m.tar.gz)

**Comment by tails1154** [comment]
Snappy seems to have a lot of issues

**Comment by dupsneerg** [comment]
> Snappy seems to have a lot of issues

Yeah, it seems so.  I'm running a pretty minimal install, and the only papercut that actually hurts a little is this one.  Cheers!

**Comment by cu-glas** [comment]
Using syntax7 did not help with CAROLINE board running LMDE. Log files can be found in the issue I created (#208).

**Comment by dupsneerg** [comment]
There is joy on SNAPPY with Arch!
I executed `sudo pacman -Syu` and kernel upgraded to 6.12.8-arch1-1.
I pulled in the script anew using `git clone https://github.com/WeirdTreeThing/chromebook-linux-audio` and ran it with the parameters `./setup-audio --force-avs-install --branch syntax-7` and the audio is working as expected.
I've uploaded an updated log.  Thank you so much for the work!
[debug-logs-Snappy-2025-01-03_23h10m.tar.gz](https://github.com/user-attachments/files/18305503/debug-logs-Snappy-2025-01-03_23h10m.tar.gz)

**Comment by avm99963** [comment]
I want to reaffirm what @dupsneerg said about updating the Kernel. In my
Pixelbook (also running Arch Linux), apart from this issue, there was a
bug in the Kernel which also prevented the audio from working:
https://bugzilla.kernel.org/show_bug.cgi?id=219577.

**Comment by SpieringsAE** [comment]
running `./setup-audio --branch syntax-7` fixes the issue on DRATINI running arch, nice!

**Comment by unamlosito** [comment]
Dragonair fedora, issue not completely solved with ```./setup-audio --branch syntax-7```.
System does recognize speakers, but only for some seconds after bootup, later on, I can't adjust volume if not with alsamixer.

https://github.com/user-attachments/assets/964ed119-c2b2-445e-9f21-dfb5121239b7


https://github.com/user-attachments/assets/912154de-bec3-4872-93bc-cb54a5720aba


https://github.com/user-attachments/assets/91095dcd-8481-4720-91ee-f9a209d514c5

**Comment by ShawHaines** [comment]
Had the same issue and `./setup-audio --branch syntax-7` fixes it for me. 
Asus CX9 (CX9400) Boardname `DROBIT`
Fedora 40 kernel version `6.12.7-100.fc40.x86_64`

**Comment by cameronj86** [comment]
`./setup-audio --branch syntax-7` worked for ELDRID running kernel 6.12.9 Debian Trixie

**Comment by WeirdTreeThing** [comment]
I've merged syntax-7 into the main ucm branch, so this is no longer required. If syntax 7 breaks anything for you, you can use the old-syntax branch instead.

**Comment by Integral-Tech** [comment]
@WeirdTreeThing After updating alsa-ucm-conf to 1.2.14, pacman complains about some conflicting file:
```
error: failed to commit transaction (conflicting files)
alsa-ucm-conf: /usr/share/alsa/ucm2/conf.d/acp3xalc5682101/acp3xalc5682101.conf exists in filesystem
Errors occurred, no packages were upgraded.
```

**Comment by WeirdTreeThing** [comment]
This is because I've been working on upstreaming the UCM. You can just remove the conflicting file and update again.
